docs/source/daml/intro/13_Interfaces.rst

```diff
@@ -0,0 +1,252 @@
+templates by the Daml Script tests ``cashTest`` and ``nftTest``, respectively,
+both defined in ``intro13/daml/Main.daml``.
+
+But that's a lot of duplication! ``cashTest`` and ``nftTest`` only differ in
+the line that creates the original asset and in the names of the choices used.
+With our new interfaces ``IAsset`` and ``IAssetTransferProposal``, we can write
+the body of this test a single time, which we name ``mkAssetTest``,
+
+.. literalinclude:: daml/daml-intro-13/daml/Main.daml
+  :language: daml
+  :start-after: -- MK_ASSET_TEST_BEGIN
+  :lines: 2
+
+The idea is that it isn't the test itself, but rather a recipe for making the
+test given some inputs - in this case, ``assetTxt`` (a label used for
+debugging), ``Parties {..}`` (a structure containing the ``Party`` values for
+``Alice`` and friends) and finally ``mkAsset`` (a function that returns a
+contract value of type ``t`` when given two ``Party`` arguments - the constraint
+``Implements t IAsset`` means that ``t`` must be some template with an interface
+instance for ``IAsset``).
+
+Before looking at the body of ``mkAssetTest``, notice how we use it to define
+the new tests ``cashAssetTest`` and ``nftAssetTest``; these are almost identical
+except for the label and function given in each case to ``mkAssetTest``. In
+effect, we have abstracted those away, so we don't need to include those details
+in the body of ``mkAssetTest``:
+
+.. literalinclude:: daml/daml-intro-13/daml/Main.daml
+  :language: daml
+  :start-after: -- CASH_ASSET_TEST_BEGIN
+  :end-before: -- CASH_ASSET_TEST_END
+
+.. literalinclude:: daml/daml-intro-13/daml/Main.daml
+  :language: daml
+  :start-after: -- MK_CASH_BEGIN
+  :end-before: -- MK_CASH_END
+
+.. literalinclude:: daml/daml-intro-13/daml/Main.daml
+  :language: daml
+  :start-after: -- NFT_ASSET_TEST_BEGIN
+  :end-before: -- NFT_ASSET_TEST_END
+
+.. literalinclude:: daml/daml-intro-13/daml/Main.daml
+  :language: daml
+  :start-after: -- MK_NFT_BEGIN
+  :end-before: -- MK_NFT_END
+
+In turn, ``mkAssetTest`` isn't very different from other Daml ``Script``\s you
+might have written before: it uses ``do`` notation as usual, including
+``submit`` blocks constructed from ``Command``\s that define the ordered
+transactions that take place in the test. The main difference is that when
+querying values of interface types we cannot use the functions ``query`` and
+``queryContractId``; instead, we must use ``queryInterface``  (for obtaining the
+set of visible active contracts of a given interface type) and
+``queryInterfaceContractId`` (for obtaining a single contract given its
+``ContractId``). Importantly, these functions return the *view* of the contract
+corresponding to the used interface, rather than the contract record itself.
+This is because the ledger might contain contracts of template types that we
+don't know about but that do implement our interface, so the view is the only
+sensible thing it can see.
```

Comment:
```suggestion
This is because the ledger might contain contracts of template types that you
don't know about but that do implement our interface, so the view is the only
sensible thing it can see.
```
Does "it" refer to the ledger here?

Reply:
yes, I'll try to rephrase